tool/grind.dart

```diff
@@ -813,26 +817,38 @@ Future serveDartdocFlutterPluginDocs() async {
       pluginPackageDocsDir.path, 8005, 'serve-dartdoc-flutter-plugin-docs');
 }
 
-@Task('Build docs for a package that requires flutter with remote linking')
-buildDartdocFlutterPluginDocs() async {
+Future<WarningsCollection> _buildDartdocFlutterPluginDocs() async {
   FlutterRepo flutterRepo = await FlutterRepo.fromExistingFlutterRepo(
       await cleanFlutterRepo, 'docs-flutter-plugin');
 
-  await flutterRepo.launcher.runStreamed(
-      Platform.resolvedExecutable,
-      [
-        '--checked',
-        pathLib.join(Directory.current.path, 'bin', 'dartdoc.dart'),
-        '--link-to-remote',
-        '--output',
-        pluginPackageDocsDir.path
-      ],
-      workingDirectory: pluginPackage.path);
+  return jsonMessageIterableToWarnings(
+      await flutterRepo.launcher.runStreamed(
+          Platform.resolvedExecutable,
+          [
+            '--checked',
+            pathLib.join(Directory.current.path, 'bin', 'dartdoc.dart'),
+            '--json',
+            '--link-to-remote',
+            '--output',
+            pluginPackageDocsDir.path
+          ],
+          workingDirectory: pluginPackage.path),
+      pluginPackageDocsDir.path,
+      defaultPubCache,
+      'HEAD');
+}
+
+@Task('Build docs for a package that requires flutter with remote linking')
+buildDartdocFlutterPluginDocs() async {
+  await _buildDartdocFlutterPluginDocs();
 }
 
 @Task('Verify docs for a package that requires flutter with remote linking')
-@Depends(buildDartdocFlutterPluginDocs)
 testDartdocFlutterPlugin() async {
+  WarningsCollection warnings = await _buildDartdocFlutterPluginDocs();
+  if (!warnings.warningKeyCounts.isEmpty) {
```

Comment:
Perhaps add a `warnings.hasWarnings` getter?

Reply:
since we're referencing warningKeyCounts to print it immediately after this, I think it's clearer to not have an extra getter.